pkg/cmd/cli/version/version.go

```diff
@@ -79,6 +80,18 @@ func printVersion(w io.Writer, clientOnly bool, kbClient kbclient.Client, server
 		return
 	}
 
+	serverSemVer := semver.MajorMinor(serverStatus.Status.ServerVersion)
+	cliSemVer := semver.MajorMinor(buildinfo.Version)
+	if serverSemVer != cliSemVer {
```

Comment:
I would move this check to the end of the function, after the server version has been printed. If there is a client/server mismatch, we would return from this function without telling the user what  the server version actually was and they wouldn't know what version they need to upgrade to.

Reply:
Yeah, that makes sense.